### PR TITLE
e2e-dynamic-remotes-runtime-refactoring

### PR DESCRIPTION
**Pull request by Andranik-mikhaielian**
Screenshot 2023-01-11 at 4 52 29 PM

### CODE REVIEW

File: advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+appsData.forEach((
+    property: {
+        header: string,
+        subheader: string,
+        hostH3?: string,
+        paragraph?: string,
+        button?: string,
+        loading: string,
+        buttonHeader: string,
+        buttonH2?: string,
```
Comment (PeaceDate) [inline advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts]: isn't optional it used in both apps

File: advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        buttonParagraph: string,
+        host: number
+    }
+) => {
+    describe(`Check Host and Remote apps`, () => {
+        beforeEach(() => {
+            basePage.openLocalhost(property.host)
+        })
+        it (`Check host and Remote apps starts and running`, () => {
```
Comment (PeaceDate) [inline advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts]: You can use here `property.subHeader`
Update please to use it `Check ${property.subheader} app start and running + check app elements`

File: advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                basePage.clickElementWithText({
+                    selector: baseSelectors.button,
+                    text: String(property.button)
+                })
+            }
+            basePage.checkElementWithTextPresence({
+                selector: baseSelectors.divElement,
+                text: property.loading
+            })
+            cy.wait(2000)
```
Comment (PeaceDate) [inline advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts]: Do you really need this wait here?

File: advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            if (property.host ===3000) {
+                basePage.checkElementWithTextPresence({
+                    selector: baseSelectors.paragraph,
+                    text: getDateWithFormat('current', 'MMMM Do YYYY, h:mm:ss a')
+                })
+            }
+            else basePage.checkElementWithTextPresence({
+                selector: baseSelectors.paragraph,
+                text: getDateWithFormat('current', 'MMMM Do YYYY, h:mm:ss a')
+            })
```
Comment (PeaceDate) [inline advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts]: I think you don't need to similar methods one in if second in else
You can left one of them here

**Review by PeaceDate — DISMISSED** [review]

File: advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        buttonHeader: string,
+        buttonH2: string,
+        buttonParagraph: string,
+        host: number
+    }
+) => {
+    describe(`Check Host and Remote apps`, () => {
+        beforeEach(() => {
+            basePage.openLocalhost(property.host)
+        })
```
Comment (edchai) [inline advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts:48]: blankline after beforeEach

File: advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        it (`Check ${property.subheader} app elements and application starts and running`, () => {
```
Comment (edchai) [inline advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts]: please provide less vague name then "starts and running"

**Review by edchai — CHANGES_REQUESTED** [review]

File: advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+        hostH3: Constants.elementsText.dynamicSystemRemotesRuntime.host.hostH3,
+        paragraph: Constants.elementsText.dynamicSystemRemotesRuntime.paragraph,
+        button: Constants.elementsText.dynamicSystemRemotesRuntime.host.button,
+        loading: Constants.elementsText.dynamicSystemRemotesRuntime.loading,
+        buttonHeader: Constants.elementsText.dynamicSystemRemotesRuntime.buttonHeader,
+        buttonH2: Constants.elementsText.dynamicSystemRemotesRuntime.buttonH2,
+        buttonParagraph: Constants.elementsText.dynamicSystemRemotesRuntime.buttonParagraph,
+        host: 3000
+    },
+
```
Comment (edchai) [inline advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts]: remove blankline

File: advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        loading: string,
+        buttonHeader: string,
+        buttonH2: string,
+        buttonParagraph: string,
+        host: number
+    }
+) => {
+    describe(`Check Host and Remote apps`, () => {
+        beforeEach(() => {
+
```
Comment (edchai) [inline advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts]: remove blankline, it should be only between beforeEach and it

File: advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+            })
+            basePage.checkElementWithTextPresence({
+                selector: baseSelectors.paragraph,
+                text: getDateWithFormat('current', 'MMMM Do YYYY, h:mm:ss a')
+            })
+        })
+    })
+})
+
+
```
Comment (edchai) [inline advanced-api/dynamic-remotes-runtime-environment-variables/e2e/checkDynamicRemotesRuntimesApps.cy.ts]: remove extra blankline

**Review by edchai — CHANGES_REQUESTED** [review]

**Review by edchai — DISMISSED** [review]

**Review by ScriptedAlchemy — DISMISSED** [review]

**Review by PeaceDate — APPROVED** [review]

**Review by edchai — APPROVED** [review]

**Pull request merged**